pytorch_tabular/tabular_datamodule.py

```diff
@@ -103,6 +103,8 @@ def update_config(self) -> None:
             self.config.output_dim = len(self.config.target)
         elif self.config.task == "classification":
             self.config.output_dim = len(self.train[self.config.target[0]].unique())
+        elif self.config.task == "ssl":
+            self.config.output_dim = self.train.shape[1] - 1
```

Comment:
should this line be :
```python
 self.config.output_dim = self.train.shape[1] - self.config.output_dim
```
?

Reply:
We cannot use 
`self.config.output_dim = self.train.shape[1] - self.config.output_dim`
because we don't know if the model will be used for classification, regression, multi-target after the self supervised task, then self.config.output_dim is not defined. I have removed the number 1 and now is 
`self.config.output_dim = self.train.shape[1]`
this number is only used in order to set the output of the ssl task, so in this case it's the dimension of the MLP that can be any number, but usually it's equal to the original dimension of the input

Reply:
makes sense... anyways we will only be using the continuous and categorical features defined by the user. So maybe we can use those to get the required output dim?

Reply:
yes it make sense, now I have updated the code setting the output_dim as len(self.config.categorical_cols) + len(self.config.continuous_cols)

Reply:
resolved